### PR TITLE
add declarations for what gatherers produce and what audits consume

### PR DESCRIPTION
**Pull request by brendankenny**
step 1 to fixing #293 and being able to selectively run gathers/audits.

Commits are much simpler separately: [first commit](https://github.com/GoogleChrome/lighthouse/pull/353/commits/df66ca5f98f8f3d7be2ef72e028c4892b3de9dee) is modifying gatherers to declare the name of the artifact they produce, [second](https://github.com/GoogleChrome/lighthouse/pull/353/commits/8c22218b4cdf6296a9a8d8887052c1d9071e0b9e) is audits declaring what they consume.

### CODE REVIEW

File: src/audits/accessibility/aria-valid-attr.js [hunk 1]
```diff
@@ -43,6 +43,15 @@ class ARIAValidAttr extends Audit {
   }
 
   /**
+   * @return {!Array<string>}
```
Comment (paulirish) [inline src/audits/accessibility/aria-valid-attr.js]: The signal to noise ratio in the audit files is already pretty low.

Can you try condensing these a bit?

e.g.

#### before:
```js
  /**
   * @return {!Array<string>}
   */
  static requiredArtifacts() {
    return [
      'accessibility'
    ];
  }
```
#### after:
```js
  /**
   * @return {!Array<string>}
   */
  static requiredArtifacts() { return ['accessibility']; }
```

#### or even
```js
  static /** {!Array<string>} */ requiredArtifacts() { 
    return ['accessibility']; 
  }
```

Comment (brendankenny) [inline src/audits/accessibility/aria-valid-attr.js]: ha, I actually prefer the nicely spaced version. Easy to see it's an array, etc. The second isn't the worst, though.

Comment (paulirish) [inline src/audits/accessibility/aria-valid-attr.js]: > The second isn't the worst, though.

k i'd be super happy for that, then.

Comment (brendankenny) [inline src/audits/accessibility/aria-valid-attr.js]: done. not quite as compact...we require line breaks after curly brackets. Compromise! :D

**Comment by paulirish** [comment]
groovy. lgtm

**Comment by samthor** [comment]
Mechanically looks fine but the Closure typedefs probably need updating (although I think they're missing `httpRedirects` already for instance). you can ignore that for now though.

LGTM

**Pull request merged**